From 6cba16a610adf7eddd17113d283b3a989e26328c Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Mon, 20 Sep 2021 08:26:54 +0100
Subject: [PATCH 1/2] introspection: Pin the API version for now

---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 82f663f5..79c6cf98 100644
--- a/meson.build
+++ b/meson.build
@@ -13,7 +13,8 @@ version_array = cheese_version.split('.')
 cheese_major_version = version_array[0].to_int()
 cheese_minor_version = version_array[1].to_int()
 
-cheese_api_version = cheese_major_version.to_string() + '.0'
+#cheese_api_version = cheese_major_version.to_string() + '.0'
+cheese_api_version = 3.0
 
 cheese_prefix = get_option('prefix')
 cheese_bindir = get_option('bindir')
-- 
2.31.1


From 760fe75d06831b3d0f3d0e820f32967602878e08 Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Mon, 20 Sep 2021 08:31:58 +0100
Subject: [PATCH 2/2] Fix the API version

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 79c6cf98..48de79d1 100644
--- a/meson.build
+++ b/meson.build
@@ -14,7 +14,7 @@ cheese_major_version = version_array[0].to_int()
 cheese_minor_version = version_array[1].to_int()
 
 #cheese_api_version = cheese_major_version.to_string() + '.0'
-cheese_api_version = 3.0
+cheese_api_version = '3.0'
 
 cheese_prefix = get_option('prefix')
 cheese_bindir = get_option('bindir')
-- 
2.31.1

