diff -up cheese-2.27.90/data/cheese.schemas.in.fix-schemas cheese-2.27.90/data/cheese.schemas.in
--- cheese-2.27.90/data/cheese.schemas.in.fix-schemas	2009-08-14 15:54:43.643811926 -0400
+++ cheese-2.27.90/data/cheese.schemas.in	2009-08-14 15:54:48.860844990 -0400
@@ -148,6 +148,7 @@
       </locale>
     </schema>
 
+    <schema>
       <key>/schemas/apps/cheese/wide_mode</key>
       <applyto>/apps/cheese/wide_mode</applyto>
       <owner>cheese</owner>
